### PR TITLE
📝  Add Natspec comments for clarity

### PR DESCRIPTION
**Pull request by Aboudjem**
# Summary

Added NATSPEC comments to missing functions and updated the existing ones that were incomplete

Related Issue: #SMA-98

## Change Type

- [X] Documentation Update

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,7 +51,6 @@
     "@typescript-eslint/eslint-plugin": "^5.30.5",
     "@typescript-eslint/parser": "^5.30.5",
     "chai": "^4.3.7",
-    "dotenv": "^16.0.1",
```
Comment (ankurdubey521) [inline package.json:54]: why did we remove `dotenv`?

**Comment by ankurdubey521** [comment]
Thanks a lot for this @Aboudjem !

**Review by livingrockrises — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (Aboudjem) [reply package.json:54]: I didn't remove it, it was duplicated on devdep

Comment (ankurdubey521) [reply package.json:54]: Got it

**Pull request merged**